aviatrix/resource_aviatrix_spoke_gateway.go

```diff
@@ -473,6 +480,14 @@ func resourceAviatrixSpokeGatewayCreate(d *schema.ResourceData, meta interface{}
 		gateway.SingleAzHa = "disabled"
 	}
 
+	enableBgp := d.Get("enable_bgp").(bool)
+	if enableBgp {
+		if !goaviatrix.IsCloudType(gateway.CloudType, goaviatrix.AWS) && !goaviatrix.IsCloudType(gateway.CloudType, goaviatrix.Azure) {
```

Comment:
Check the cloud type at the same time with `!goaviatrix.IsCloudType(gateway.CloudType, goaviatrix.AWS|goaviatrix.Azure)`